contracts/VaultV3.vy

```diff
@@ -289,13 +289,43 @@ def withdraw(_shares: uint256, _recipient: address, _strategies: DynArray[addres
 
     amount: uint256 = self._amountForShares(shares)
 
-    # TODO: withdraw from strategies
-
-    assert self.totalIdle >= amount, "insufficient total idle"
+    # load to memory to save gas
+    currTotalIdle: uint256 = self.totalIdle
+    currTotalDebt: uint256 = self.totalDebt
```

Comment:
you may not need to read this here but within the `if amount > currTotalIdle`

in case of enough totalIdle before withdrawing this will spend unnecessary gas